src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_load_store_emitters.cpp

```diff
@@ -25,6 +25,89 @@ namespace ov::intel_cpu::aarch64 {
 using jit_generator = dnnl::impl::cpu::aarch64::jit_generator;
 using cpu_isa_t = dnnl::impl::cpu::aarch64::cpu_isa_t;
 
+// Helper function to load with large offset handling
+template <typename RegType>
+static void load_with_offset_check(jit_generator* h, const RegType& dst, const XReg& src, int offset) {
+    if constexpr (std::is_same_v<RegType, VReg> || std::is_same_v<RegType, QReg>) {
+        // Manual offset handling for VReg/QReg due to uni_ldr limitations
+        const int off_mod = offset % 16;
+        const int off_mul_vl = offset / 16;
+
+        if (off_mod == 0 && off_mul_vl >= 0 && off_mul_vl <= 4095) {
+            h->ldr(QReg(dst.getIdx()), ptr(src, static_cast<uint32_t>(offset)));
+        } else {
+            h->add_imm(h->X_DEFAULT_ADDR, src, offset, h->X_TMP_0);
+            h->ldr(QReg(dst.getIdx()), ptr(h->X_DEFAULT_ADDR));
+        }
+    } else {
+        // Manual offset handling for other register types
+        // Note: read about LDR (immediate) in the manual Arm A-profile A64 Instruction Set Architecture
+        int max_offset = 4095;  // Default fallback
+        int alignment = 1;      // Default fallback
+        if constexpr (std::is_same_v<RegType, SReg>) {
+            max_offset = 16380;
+            alignment = 4;
+        } else if constexpr (std::is_same_v<RegType, DReg>) {
+            max_offset = 32760;
+            alignment = 8;
+        } else if constexpr (std::is_same_v<RegType, HReg>) {
+            max_offset = 8190;
+            alignment = 2;
+        } else if constexpr (std::is_same_v<RegType, BReg>) {
+            max_offset = 4095;
+            alignment = 1;
+        }
+
+        if (offset >= 0 && offset <= max_offset && (offset % alignment) == 0) {
+            h->ldr(dst, ptr(src, static_cast<uint32_t>(offset)));
+        } else {
+            h->add_imm(h->X_DEFAULT_ADDR, src, offset, h->X_TMP_0);
+            h->ldr(dst, ptr(h->X_DEFAULT_ADDR));
+        }
+    }
```

Comment:
This code looks very similar to the code on L36-L41. Is it possible to unify it somehow? For instance, return `max_offset = 4095` and `alignment = 16` for `VReg` and `QReg`
If this can be done, please modify `store_with_offset_check` in similar way